juspay/hyperswitch
# refactor(core): restrict requires_customer_action in confirm

## Type of Change



## Description

This PR will restrict a payment to be confirmed again if the status is `requires_customer_action`.




## Motivation and Context

A payment which is in `requires_customer_action` cannot be confirmed again. Only failed payments can be retried by passing `retry_action` as `retry`.

## How did you test it?

- Confirm a payment which is in `requires_customer_action` state.
<img width="982" alt="Screenshot 2024-01-03 at 7 09 41 PM" src="https://github.com/juspay/hyperswitch/assets/48803246/c74c6f0d-0cc6-465d-8ea2-66c9b682127b">

- Create sepa bank transfer payment via stripe ( payments_confirm happens via webhooks )
<img width="619" alt="Screenshot 2024-01-03 at 8 46 11 PM" src="https://github.com/juspay/hyperswitch/assets/48803246/e4263ddc-3f24-436a-bc95-ba1c6e7fa16c">

- Payment updated by charged webhook
<img width="1404" alt="Screenshot 2024-01-03 at 8 47 38 PM" src="https://github.com/juspay/hyperswitch/assets/48803246/a322c0de-f9db-4358-bff9-b416ebb343d7">


## Checklist
